.github/workflows/lint.yaml

```diff
@@ -0,0 +1,27 @@
+name: Lint source code
+on:
+  - push
```

Comment:
is style check happen on any branch after commit? I believe yes, just wanted to confirm.

Reply:
Any time a branch is pushed it will trigger the workflow.

I've been using `push` by default during development for all workflows, because otherwise I can't test it. But it can definitely be changed before merging.

Reply:
I think it would be better if we trigger jobs based on the PR request and main branch direct push only. Otherwise unnecessarily lot of artifacts would be generated after every push. What do you think?